docs/docs/internal/xcmp/how-to-open-xcmp-channel.md

```diff
@@ -0,0 +1,147 @@
+
+Chain accounts can be obtained by mapping parachain id to account via [substrate-js-utilities] 
+
+
+## Send request open channel to counterparty parachain
+
+Example, [0x290001010002100004000000000b0060defb740513000000000b0060defb74050006000700f2052a01383c00e8030000e8030000009001000d0100040001009d20](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fpicasso-rpc.composable.finance#/extrinsics/decode/0x290001010002100004000000000b0060defb740513000000000b0060defb74050006000700f2052a01383c00e8030000e8030000009001000d0100040001009d20).
+
+If will request open `hrmp` to other parachain
+
+It will execute request to open channel (half of future duplex).
+
+**Details**
+
+Message will be sent to Relay and will pay all fees in relay native token.
+
+Ensure transaction executed from `Native` (Parachain itself) `originType`
+ 
+Ensure `DepositAsset` will deposit to relevant Parachain identifier (spending chain)
+
+Ensure `requireWeightAtMost` is safely larger (e.g. 2-3x) then
+
+`Developer -> Runtime Calls -> TransactionPaymentCallApi -> queryCallInfo -> weight`
+but strictly (and if possible safely) less then. 
+
+`requireWeightAtMost` should be equivalent less than than in `BuyExecution` (you can compare it to `queryCallInfo.partialFee` of call and make `BuyExecution` 10x+ larger than this). In case of failure to do so XCM will fail with `TooExpensive`
+
+Do not set `requireWightAtMost` to big as you will get `dmpQueue.OverweightEnqueued`. So  `requireWightAtMost` has some of semantics of `requireWeightMinimum`. It is not clear if message will be retried automatically. For sanity check `configuration -> activeConfig -> umpMaxIndividualWeight` is way higher then our transact.
+
+For safety can consider scheduling transact message instead of executing immediately.
+
+[0x3c00e8030000e803000000900100](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.polkadot.io#/extrinsics/decode/0x3c00e8030000e803000000900100).
+
+Check `Transact` message has proper existing parachain recipient.
+
+`proposedMaxCapacity` is less or equal to `Storage -> configuration -> activeConfig` `hrmpChannelMaxCapacity`
+
+`proposedMaxMessageSize` is less or equal to `hrmpChannelMaxMessageSize` here.
+
+`hrmpMaxParachain*Channels` is not exceeded by neither participant chains.
+
+## Counterparty chain to accept our request and request back
+
+Encode above preimage via `Democracy -> Submit preimage`.
+
+Then `Submit proposal`.
+
+https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkarura-rpc-3.aca-api.network%2Fws#/extrinsics/decode/0x33000101000210000400000000070010a5d4e81300000000070010a5d4e800060003009435775c1802083c01270800003c0027080000e8030000009001000d010004000100411f
+
+**Details** 
+
+Find latest `proposal` or success `referenda` to open channels and compare difference with our message. Things change, so further example away more likely it is not up to date.
+
+Please do all relevant checks from previous step.
```

Comment:
```suggestion
**Details** 

Find the latest `proposal` or successful `referenda` to open channels and compare differences with our message. Things change so the further examples are likely to be out of date.

Please do all relevant checks from the previous step.
```
Ambiguous; Is the assumption in my suggestion correct or could you elaborate on the examples being out of date?

Reply:
yes.

Reply:
i will remove this lines - just wanted to say there is much details and churn - so this guide is no 100% exact at time one may read it.